app/src/main/java/com/bashkevich/androidfundamentals/model/notifications/MovieNotifications.kt

```diff
@@ -0,0 +1,95 @@
+package com.bashkevich.androidfundamentals.model.notifications
+
+import android.app.PendingIntent
+import androidx.core.app.NotificationChannelCompat
+import android.content.Context
+import android.content.Intent
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.graphics.drawable.BitmapDrawable
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.core.app.NotificationManagerCompat.IMPORTANCE_HIGH
+import androidx.core.net.toUri
+import coil.ImageLoader
+import coil.request.ImageRequest
+import com.bashkevich.androidfundamentals.R
+import com.bashkevich.androidfundamentals.model.viewobject.Movie
+import java.net.URI
+import java.net.URL
+
+
+class MovieNotifications(private val appContext: Context) {
+    private val notificationManagerCompat = NotificationManagerCompat.from(appContext)
+
+    fun initialize() {
+        if (notificationManagerCompat.getNotificationChannel(CHANNEL_NEW_MOVIES) == null) {
+
+            val notificationChannel =
+                NotificationChannelCompat.Builder(CHANNEL_NEW_MOVIES, IMPORTANCE_HIGH)
+                    .setName(appContext.getString(R.string.notification_channel_name))
+                    .setDescription(
+                        appContext.getString(
+                            R.string.notification_channel_description
+                        )
+                    ).build()
+
+            notificationManagerCompat.createNotificationChannel(notificationChannel)
+        }
+    }
+
+    fun showMovieNotification(movie: Movie) {
+        val contentUri = "$BASE_CONTENT_URI${movie.id}".toUri()
+
+        val movieIntent = Intent().setAction(Intent.ACTION_VIEW).setData(contentUri)
+
+        val pendingIntent = PendingIntent.getActivity(
+            appContext,
+            REQUEST_CODE,
+            movieIntent,
+            PendingIntent.FLAG_UPDATE_CURRENT
+        )
+
+        val testBitmap =
+            BitmapFactory.decodeStream(URL(movie.poster).openConnection().getInputStream())
+
+        val notification = NotificationCompat.Builder(appContext, CHANNEL_NEW_MOVIES)
+            .setSmallIcon(R.drawable.ic_baseline_local_movies_24)
+            .setLargeIcon(testBitmap)
+            .setContentTitle(movie.title)
+            .setContentText(appContext.getString(R.string.notification_description, movie.ratings))
+            .setPriority(NotificationCompat.PRIORITY_HIGH).setWhen(System.currentTimeMillis())
+            .setStyle(
+                NotificationCompat.BigPictureStyle()
+                    .bigPicture(testBitmap)
+                    .bigLargeIcon(null)
+            )
+            .setAutoCancel(true).setContentIntent(pendingIntent)
+
+        val loader = ImageLoader(appContext)
```

Comment:
Is it used?

Reply:
Not yet, but I kept it to change image loading

Reply:
I don't think you need to keep code that you can google or find in the git history.